[Enhancement] - ServiceNow - Non-Interactive OAuth2 for v2 Integration (#42975)
demisto/content: Packs/ServiceNow/Integrations/ServiceNow_CMDB/README.md

To use OAuth 2.0 authorization, perform the following steps:

1. Login to your ServiceNow instance and create an endpoint for XSOAR to access your instance (please see [Snow OAuth](https://docs.servicenow.com/bundle/xanadu-platform-security/page/administer/security/concept/c_OAuthApplications.html) for more information).

> 2. Copy the **Client Id** and **Client Secret** (press the lock next to the client secret to reveal it) that were automatically generated when creating the endpoint into the **Client ID** and **Client Secret** fields of the instance configuration.

Reviewer: ```suggestion
2. Copy the **Client Id** and **Client Secret** (click the lock next to the client secret to reveal it) that were automatically generated when creating the endpoint into the **Client ID** and **Client Secret** fields of the instance configuration.
```

> 3. (Recommended) Enter the ServiceNow account's **Username** and **Password** in the instance configuration. When provided, the integration will automatically perform the initial OAuth login on first use — no manual ***!servicenow-cmdb-oauth-login*** command is needed. It will also automatically renew the refresh token when it expires.

Reviewer: ```suggestion
3. (Recommended) Enter the ServiceNow account **Username** and **Password** in the instance configuration. When provided, the integration automatically performs the initial OAuth login on first use. It also automatically renews the refresh token when it expires.
```

4. Select the **Use OAuth Login** checkbox and click **Done**.
5. If you did **not** provide **Username** and **Password** in step 3, run the command ***!servicenow-cmdb-oauth-login*** from the XSOAR CLI and fill in the username and password of the ServiceNow instance. This step generates and saves to the integration context a refresh token to the ServiceNow instance and is required only the first time after configuring a new instance in the XSOAR platform. If you provided **Username** and **Password**, this step is handled automatically.

> 6. (Optional) Test the created instance by running the ***!servicenow-cmdb-oauth-test*** command.

Reviewer: ```suggestion
6. (Optional) Test the created instance by running the `!servicenow-cmdb-oauth-test` command.
```

**Notes:**

1. When running the ***!servicenow-cmdb-oauth-login*** command, a refresh token is generated and will be used to produce new access tokens after the current access token has expired.

> 2. If the **Username** and **Password** fields are configured, the integration will automatically perform the initial login and renew the refresh token when it expires — no manual commands are needed. Otherwise, you will have to run the ***servicenow-cmdb-oauth-login*** command for the initial login and again each time the refresh token expires. We recommend setting the **Refresh Token Lifespan** field in the endpoint created in step 1 to a long period (can be set to several years).

Reviewer: ```suggestion
2. If the **Username** and **Password** fields are configured, the integration automatically performs the initial login and renews the refresh token when it expires. Otherwise, you have to run the `servicenow-cmdb-oauth-login` command for the initial login and again each time the refresh token expires. We recommend setting the **Refresh Token Lifespan** field in the endpoint created in step 1 to a long period (can be set to several years).
```
